miio/huizuo.py

```diff
@@ -66,17 +137,84 @@ def color_temp(self) -> int:
         """Return current color temperature."""
         return self.data["color_temp"]
 
+    @property
+    def is_fan_on(self) -> Optional[bool]:
+        """Return True if Fan is on."""
+        if "fan_power" in self.data:
+            return self.data["fan_power"]
+        return None
+
+    @property
+    def fan_speed_level(self) -> Optional[int]:
+        """Return current Fan speed level."""
+        if "fan_level" in self.data:
+            return self.data["fan_level"]
+        return None
+
+    @property
+    def is_fan_reverse(self) -> Optional[bool]:
+        """Return True if Fan reverse is on."""
+        if "fan_motor_reverse" in self.data:
+            return self.data["fan_motor_reverse"]
+        return None
+
+    @property
+    def fan_mode(self) -> Optional[int]:
+        """Return 0 if 'Basic' and 1 if 'Natural wind'"""
+        if "fan_mode" in self.data:
+            return self.data["fan_mode"]
+        return None
+
+    @property
+    def is_heater_on(self) -> Optional[bool]:
+        """Return True if Heater is on."""
+        if "heater_power" in self.data:
+            return self.data["heater_power"]
+        return None
+
+    @property
+    def heater_fault_code(self) -> Optional[int]:
+        """Return Heater's fault code. 0 - No Fault"""
+        if "heater_fault_code" in self.data:
+            return self.data["heater_fault_code"]
+        return None
+
+    @property
+    def heat_level(self) -> Optional[int]:
+        """Return Heater's heat level"""
+        if "heat_level" in self.data:
+            return self.data["heat_level"]
+        return None
+
     def __repr__(self):
-        s = "<Huizuo on=%s brightness=%s color_temp=%s>" % (
-            self.is_on,
-            self.brightness,
-            self.color_temp,
-        )
+        parameters = []
+        if self.is_on is not None:
+            parameters.append("on=" + self.is_on)
+        if self.brightness is not None:
+            parameters.append("brightness=" + self.brightness)
+        if self.color_temp is not None:
+            parameters.append("color_temp=" + self.color_temp)
+        if self.is_fan_on is not None:
+            parameters.append("fan_on=" + self.is_fan_on)
+        if self.fan_speed_level is not None:
+            parameters.append("fan_level=" + self.fan_speed_level)
+        if self.fan_mode is not None:
+            parameters.append("fan_mode=" + self.fan_mode)
+        if self.is_fan_reverse is not None:
+            parameters.append("fan_motor_reverse=" + self.is_fan_reverse)
+        if self.is_heater_on is not None:
+            parameters.append("heater_on=" + self.is_heater_on)
+        if self.heat_level is not None:
+            parameters.append("heat_level=" + self.heat_level)
+        if self.heater_fault_code is not None:
+            parameters.append("heater_fault_code=" + self.heater_fault_code)
```

Comment:
Maybe creating a list of available properties, and using `getattr(self, 'property_name')` in a loop to avoid repetition and make the code cleaner?

So something like this, although there is probably a nicer way to get a list of all properties (to avoid the string list):
```
optional_props = ['is_fan_on', ..., 'fan_speed_level']
for prop in optional_props:
    val = getattr(self, prop)
    if val is not None:
        parameters.append(f"{prop}={val}")
```

Reply:
Changed as you suggested.
I added all available properties to the list of properties to have one approach to describe the properties